controllers/machinehealthcheck_controller.go

```diff
@@ -151,7 +175,50 @@ func (r *MachineHealthCheckReconciler) reconcile(_ context.Context, cluster *clu
 		UID:        cluster.UID,
 	})
 
-	return ctrl.Result{}, fmt.Errorf("controller not yet implemented")
+	logger := r.Log.WithValues("machinehealthcheck", m.Name, "namespace", m.Namespace)
+	logger = logger.WithValues("cluster", cluster.Name)
+
+	// Create client for target cluster
+	clusterClient, err := remote.NewClusterClient(r.Client, cluster, r.scheme)
+	if err != nil {
+		logger.Error(err, "Error building target cluster client")
+		return ctrl.Result{}, err
+	}
+
+	err = r.watchClusterNodes(ctx, r.Client, cluster)
+	if err != nil {
+		logger.Error(err, "Error watching nodes on target cluster")
+		return ctrl.Result{}, err
+	}
+
+	// fetch all targets
+	logger.V(3).Info("Finding targets", "request", namespacedName(m))
```

Comment:
how would the end logging message look like here? wouldn't this be redundant with the info given by https://github.com/kubernetes-sigs/cluster-api/pull/https://github.com/go-logr/logr/blob/HEAD/logr.go#diff-31e38a3ed71e72141d235f4892dcd9f5R178 i.e the mhc we are operating over.

Reply:
Your link is messed up, but yes, namespacedName(m) is the same as the values for machinehealthcheck and namespace on line 178

Reply:
I'll get rid of this parameter where it's been used, I agree it's redundant